bin/mango-notebook

```diff
@@ -47,29 +86,55 @@ fi
 # get Mango jar
 MANGO_CLI_JAR=$(${SOURCE_DIR}/find-mango-assembly.sh)
 
-SPARK_SUBMIT=$(${SOURCE_DIR}/find-spark.sh)
-echo "Using spark-submit=$SPARK_SUBMIT" 1>&2
-
-if [ -z "$SPARK_HOME" ]; then
-  PYSPARK=$(which pyspark || echo)
+# get pyspark path
+if [[ -z "${SPARK_HOME}" ]]; then
+  wrapper_path=$(get_program_path spark-script-wrapper.sh false)
+  if [[ -f "${wrapper_path}" ]]; then
+    . ${wrapper_path}
+    PYSPARK=$(find_script pyspark)
+  else
+    PYSPARK=$(get_program_path pyspark)
+  fi
+  SPARK_HOME=${PYSPARK/bin\/pyspark}
 else
-  PYSPARK="$SPARK_HOME"/bin/pyspark
+  PYSPARK="${SPARK_HOME}"/bin/pyspark
 fi
-if [ -z "$PYSPARK" ]; then
-  echo "SPARK_HOME not set and spark-shell not on PATH; Aborting."
+
+if [[ ! -f "${PYSPARK}" ]]; then
+  print_err "pyspark was not found on PATH and SPARK_HOME; Aborting."
   exit 1
 fi
 
-export PYSPARK_PYTHON=$(which python || echo)
-export PYSPARK_DRIVER_PYTHON=jupyter
-export PYSPARK_DRIVER_PYTHON_OPTS="notebook $NOTEBOOK_ARGS"
+# Use standard PYSPARK environement variables
+if [[ -z ${PYSPARK_PYTHON} ]]; then
+  export PYSPARK_PYTHON=$(get_program_path python)
+fi
+
+if [[ -z ${PYSPARK_DRIVER_PYTHON} ]]; then
+  export PYSPARK_DRIVER_PYTHON=jupyter
+fi
+
+if [[ -z ${PYSPARK_DRIVER_PYTHON_OPTS} ]]; then
+  export PYSPARK_DRIVER_PYTHON_OPTS="notebook ${NOTEBOOK_ARGS}"
+else
+  export PYSPARK_DRIVER_PYTHON_OPTS="${PYSPARK_DRIVER_PYTHON_OPTS} ${NOTEBOOK_ARGS}"
+fi
+
+if [[ -d "${SPARK_HOME}/python/lib" ]]; then
+  PY4J_ZIP_ALL_VERSIONS=( ${SPARK_HOME}/python/lib/py4j-+([0-9]).+([0-9]).+([0-9])-src.zip )
+  # use the latest version
+  PY4J_ZIP="${PY4J_ZIP_ALL_VERSIONS[-1]}"
+  if [[ ! -e ${PY4J_ZIP} ]]; then
+    print_err "${PY4J_ZIP} is not found; Aborting."
+    exit 1
+  fi
+fi
 
-PY4J_ZIP="$(ls -1 "${SPARK_HOME}/python/lib" | grep py4j)"
```

Comment:
We need access to this zip to run correctly on workers. Is there a reason it is removed?